src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -346,6 +334,57 @@ public virtual VerifyResult Verify(StoreView snapshot, BigInteger totalSenderFee
             return VerifyResult.Succeed;
         }
 
+        #region Oracles
+
+        private void LoadOracleAttribute()
+        {
+            if (oracleCached != OracleCache.None)
+            {
+                return;
+            }
+
+            var attr = attributes.Where(u => u.Usage == TransactionAttributeUsage.OracleRequest).FirstOrDefault();
+
+            if (attr != null)
+            {
+                oracleCached |= OracleCache.Request;
+            }
+
+            attr = attributes.Where(u => u.Usage == TransactionAttributeUsage.OracleResponse).FirstOrDefault();
+
+            if (attr != null)
+            {
+                oracleCached |= OracleCache.Response;
+                oracleResponse = attr.Data.AsSerializable<OracleResponseAttribute>();
+            }
+            else
+            {
+                oracleResponse = null;
+            }
+        }
+
+        public bool IsOracleResponse(out UInt256 requestTx)
+        {
+            LoadOracleAttribute();
```

Comment:
Move `LoadOracleAttribute` into `Deserialize`?

Reply:
We can call it in Deserialize, but it could be changed outside